extensions/vertx-http/deployment-spi/src/main/java/io/quarkus/vertx/http/deployment/spi/GeneratedStaticResourceBuildItem.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.vertx.http.deployment.spi;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * GeneratedStaticResourceBuildItem aims to be used by extensions to generated static resources into
+ * <code>META-INF/resource</code> directory. This means to be used just for {@link io.quarkus.runtime.LaunchMode#TEST} or
+ * {@link io.quarkus.runtime.LaunchMode#DEVELOPMENT} mode.
```

Comment:
It is not only for dev or test, it is also for production.

Could you add a bit of explanation on what it does behind the scene: It takes care of creating what needed for routing in dev, test, and prod mode and native (you don't create any other BuildItem for this to be working).